dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceQueryParams.java

```diff
@@ -277,6 +277,12 @@
      */
     private Date skipChangedBefore;
 
+    /**
+     * Indicates whether tracked entity instance is marked as a Potential
+     * Duplicate
+     */
+    private Boolean potentialDuplicate;
```

Comment:
I am not sure we have a convention for booleans so I might find examples for different styles 😅 

I think the name `potentialDuplicate` for a request parameter is a little confusing. I would rather name it `includePotentialDuplicate`. Since it should indicate whether we return such TEIs and not what you describe in the comment above. So the comment should definitely change (could you please include in the comment what `null` means. That is often not clear). I think the query param as well but someone else can also add their view.

Like we do for

- `includeDeleted`
- `includeAllAttributes`

see https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-master/new-tracker.html#request-parameters

Reply:
mmh i see what you mean but in this case, I'd stick with potentialDuplicate because it is a tei property. 
- `inlucdeDeleted` , it defaults to false because I think we are not potentially interested in deleted tei?
- `includeAllAttributes`, still defaults to false, I see it is in use to check ownership

Potential duplicate shouldn't have a default,  we may want to filter for true or false. That is why is null by default and in that case, we get all. We can add it to the doc for sure. 
Anyways, I'll wait also for others to give an opinion

Reply:
I think here `potentialDuplicate` is more like `followUp` parameter.
The naming follow the pattern of the other parameters, so I don't think it is a problem. I would improve the comment though explain exactly what it happens if it is TRUE, FALSE and null.

Reply:
By default, we return teis regardless of their "potentialDuplicate" value. That means, true or false is not the only options as Luca mentions, but also null (ignore/dont filter by this property).

I agree, @lucaCambi77 please improve the comment, stating each case of true/false/null and the intended result; That will make it clear why we are using a Boolean, and how null is significant.

I think the naming is fine; And if null was not valid in this case, I agree it should be boolean :)